src/encoder.rs

```diff
@@ -1110,27 +1110,32 @@ pub fn encode_tx_block<T: Pixel>(
   cw: &mut ContextWriter,
   w: &mut dyn Writer,
   p: usize,
+  // Offset in the luma plane of the partition enclosing this block.
   tile_partition_bo: TileBlockOffset,
-  bx: usize,
-  by: usize, // tx block position within a partition, unit: tx block number
-  tile_bo: TileBlockOffset,
+  // tx block position within a partition, unit: tx block number
+  bx: usize, by: usize,
+  // Offset in the luma plane where this tx block is colocated. Note that for
+  // a chroma block, this offset might be outside of the current partition.
+  // For example in 4:2:0, four 4x4 luma partitions share one 4x4 chroma block,
+  // this block is part of the last 4x4 partition, but its `tx_bo` offset
+  // matches the offset of the first 4x4 partition.
+  tx_bo: TileBlockOffset,
```

Comment:
Yes, this is really correct comment! The correct seg_id used by chroma 4x4 block is the seg_id for last luma 4x4.